internal/characters/albedo/burst.go

```diff
@@ -33,28 +33,55 @@ func (c *char) Burst(p map[string]int) action.ActionInfo {
 		Durability: 25,
 		Mult:       burst[c.TalentLvlBurst()],
 	}
-	snap := c.Snapshot(&ai)
 
-	//check stacks
-	if c.Base.Cons >= 2 && c.StatusIsActive(c2key) {
-		ai.FlatDmg += (snap.BaseDef*(1+snap.Stats[attributes.DEFP]) + snap.Stats[attributes.DEF]) * float64(c.c2stacks)
+	c2Count := 0
+	hasC2 := c.Base.Cons >= 2 && c.StatusIsActive(c2key)
+	// C2 damage for initial hit is calculated on burst start
+	if hasC2 {
+		c2Count = c.c2stacks
 		c.c2stacks = 0
+		ai.FlatDmg = (c.Base.Def*(1+c.Stat(attributes.DEFP)) + c.Stat(attributes.DEF)) * float64(c2Count)
 	}
 
 	// initial damage
-	// TODO: damage frame
-	c.Core.QueueAttackWithSnap(
+	c.Core.QueueAttack(
 		ai,
-		snap,
 		combat.NewCircleHitOnTargetFanAngle(c.Core.Combat.Player(), nil, 8, 120),
 		burstHitmark,
+		burstHitmark,
 	)
 
-	// Blossoms are generated on a slight delay from initial hit
-	// TODO: no precise frame data for time between Blossoms
-	if c.skillActive {
+	// A4 and fatal blossom
+	// delay fatal blossom triggering until burstHitmark because that's when it:
+	// - checks whether the skill is still active
+	// - recalculates C2 damage
+	c.Core.Tasks.Add(func() {
+		// A4
+		m := make([]float64, attributes.EndStatType)
+		m[attributes.EM] = 125
+		for _, char := range c.Core.Player.Chars() {
+			char.AddStatMod(character.StatMod{
+				Base:         modifier.NewBaseWithHitlag("albedo-a4", 600),
+				AffectedStat: attributes.EM,
+				Amount: func() ([]float64, bool) {
+					return m, true
+				},
+			})
+		}
+
+		// fatal blossom
+		if !c.skillActive {
+			return
+		}
 		ai.Abil = "Rite of Progeniture: Tectonic Tide (Blossom)"
 		ai.Mult = burstPerBloom[c.TalentLvlBurst()]
+		initialFatalBlossomHitmark := fatalBlossomHitmark - burstHitmark
```

Comment:
nit: should we just change the value of `fatalBlossomHitmark` instead? (can do `145 - burstHitmark` if we want to keep the current value for posterity)

Reply:
yeah I'll change that very soon

Reply:
done